tests/foreman/api/test_errata.py

```diff
@@ -224,26 +824,93 @@ def test_positive_install_multiple_in_host(target_sat, rhel_contenthost, module_
             search_rate=20,
             max_tries=15,
         )
-        rhel_contenthost.execute('subscription-manager repos')
-        sleep(20)
+        # Wait for async errata recalculate task(s), invoked by REX task
+        target_sat.api_factory.wait_for_errata_applicability_task(
+            host_id=rhel_contenthost.nailgun_host.id,
+            from_when=epoch_timestamp,
+        )
+        # Host Applicable Errata count decreased by one
         assert (
             rhel_contenthost.applicable_errata_count == pre_errata_count - 1
-        ), f'Host applicable errata did not decrease by one, after installation of {errata}'
+        ), f'Host applicable errata did not decrease by one, after installation of {ERRATUM}'
+        # Applying this ERRATUM updated one or more of the erratum's listed packages
+        found_updated_packages = []
+        for package in errata_packages:
+            result = rhel_contenthost.run(f'rpm -q {package}')
+            if result.status == 0:
+                assert (
+                    package in FAKE_9_YUM_UPDATED_PACKAGES
+                ), f'An unexpected package: "{package}", was updated by this errata: {ERRATUM}.'
+                if package in ERRATUM_instance.packages:
+                    found_updated_packages.append(package)
+
+        assert len(found_updated_packages) > 0, (
+            f'None of the expected errata.packages: {errata_packages}, were found on host: "{rhel_contenthost.hostname}",'
+            f' after installing the applicable errata: {ERRATUM}.'
+        )
+        # Host Applicable Packages count dropped by number of packages updated
+        assert rhel_contenthost.applicable_package_count == pre_package_count - len(
+            found_updated_packages
+        ), (
+            f'Host: "{rhel_contenthost.hostname}" applicable package count did not decrease by {len(found_updated_packages)},'
+            f' after errata: {ERRATUM} installed updated packages: {found_updated_packages}'
+        )
+        installed_errata.append(ERRATUM)
+        updated_packages.extend(found_updated_packages)
+    
+    # In case no ERRATUM in list are applicable:
+    # Lack of any package or errata install will raise `AssertionError`.
+    assert (
+        len(installed_errata) > 0
+    ), f'No applicable errata were found or installed from list: {FAKE_9_YUM_SECURITY_ERRATUM}.'
+    assert (
+        len(updated_packages) > 0
+    ), f'No applicable packages were found or installed from list: {FAKE_9_YUM_UPDATED_PACKAGES}.'
+    # Each expected erratum and packages installed only once
+    assert set(updated_packages), f'Expect no repeated packages in install list: {updated_packages}'
+    assert set(installed_errata), f'Expect no repeated errata in install list: {installed_errata}'
```

Comment:
I don't believe these asserts are checking what they're expected to (based on the custom message)

```python
In [1]: assert set([1,2,3]), "There are duplicates"

In [2]: assert set([1,2,2,3]), "There are duplicates"

In [3]: 
```

Reply:
fixed by casting to set, then asserting the length of the set is equal to length of original list. For both `updated_packages` and `installed_errata` lists, in next commit!